### PR TITLE
Modify diff colors

### PR DESCRIPTION
**Pull request by xiongtx**
Follow-up on #293.

- Make diff colors darker
- Have `ediff-*`, `magit-diff-*` inherit from `diff-*`

**Note**: the `diff-changed-*` colors have not been touched; I can't figure out a way to make the yellows fit the `zenburn-yellow{-,+}*`theme. It'd be great if someone wants to take a crack at that.

Highlighted (top, point in region) and unhighlighted (bottom):

image

3-way Ediff:

image

### CODE REVIEW

**Comment by bbatsov** [comment]
> I can't figure out a way to make the yellows fit the zenburn-yellow{-,+}*theme

I don't quite get this. The existing 3 colors don't look good or what?

**Comment by xiongtx** [comment]
The contrast is pretty awful :man_shrugging:.

**Comment by bbatsov** [comment]
I guess you can use something like color-ligthen/darken to produce darker versions of the yellow (e.g. by steps of something like 10-20%). I'm not expert in colors myself, but this seems reasonable. Or we can use orange?

**Comment by impaktor** [comment]
Either way, looks to be an improvement. 👍

**Comment by xiongtx** [comment]
> I guess you can use something like color-ligthen/darken to produce darker versions of the yellow 

I think I've had enough mucking w/ :rainbow: for now :tired_face:. Let's get this merged and address the yellow later.

**Comment by impaktor** [comment]
@xiongtx what is it that should be yellow? Is it something derived from vim's zenburn, or the original diff colours? Screenshot?

**Comment by xiongtx** [comment]
The `diff-changed-*` faces should be a certain yellow. The default ones work well w/ the `diff-added` and `diff-removed` I've implemented here.

Don't have a screenshot handy. You can play around w/ scaling the Zenburn yellows if you like.

File: zenburn-theme.el [hunk 1]
```diff
@@ -57,7 +57,7 @@
     ("zenburn-yellow"   . "#F0DFAF")
     ("zenburn-yellow-1" . "#E0CF9F")
     ("zenburn-yellow-2" . "#D0BF8F")
-    ("zenburn-green-1"  . "#5F7F5F")
+    ("zenburn-green-2"  . "#5F7F5F")
```
Comment (bbatsov) [inline zenburn-theme.el]: Hmm, I noticed this change just after I merged the PR, but I don't quite get it. What was wrong with the original name? Now it's odd we don't have `-1`.

Comment (bbatsov) [reply zenburn-theme.el]: Ah, forget about this.

**Pull request merged**